sci-libs/cantera/cantera-2.4.0.ebuild

```diff
@@ -0,0 +1,130 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+PYTHON_COMPAT=( python3_{4,5,6} )
+
+inherit desktop python-single-r1 scons-utils toolchain-funcs
+
+DESCRIPTION="Object-oriented tool suite for chemical kinetics, thermodynamics, and transport"
+HOMEPAGE="http://www.cantera.org"
+SRC_URI="https://github.com/Cantera/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="+cti cxx fortran pch +python static-libs test"
+
+REQUIRED_USE="
+	|| ( cxx python )
+	cti? ( ${PYTHON_REQUIRED_USE} )
+	fortran? ( cxx )
+	python? ( cti )
+	static-libs? ( cxx )
+	^^ ( $(python_gen_useflags 'python3*') )
+	"
+
+RDEPEND="
+	python? (
+		dev-python/numpy[${PYTHON_USEDEP}]
+	)
+"
+
+DEPEND="
+	${RDEPEND}
+	dev-cpp/eigen
+	dev-libs/boost
+	dev-libs/libfmt:0=
+	dev-util/scons[${PYTHON_USEDEP}]
+	sci-libs/sundials
+	fortran? (
+		sci-libs/sundials[fortran]
+		sys-devel/gcc[fortran]
+	)
+	python? (
+		dev-python/cython[${PYTHON_USEDEP}]
+	)
+	test? (
+		>=dev-cpp/gtest-1.8.0
+	)
+"
+
+PATCHES=( "${FILESDIR}/${PN}_${PV}_disable_debug_and_optimization.patch" )
+
+src_prepare() {
+	default
+	# modify SConstruct to set env['libdirname'] to $(get_libdir)
+	sed -i "1469,1471s/^/#/" "${S}"/SConstruct || die "failed to modify 'SConstruct'"
+	sed -i "1472s/^[ \t]*//; s/'lib'/'$(get_libdir)'/" "${S}"/SConstruct || die "failed to modify 'SConstruct' with get_libdir"
+}
+
+## Full list of configuration options of Cantera is presented here:
+## http://cantera.org/docs/sphinx/html/compiling/config-options.html
+
+src_configure() {
+	scons_vars=(
+		CC="$(tc-getCC)"
+		CXX="$(tc-getCXX)"
+		cc_flags="${CXXFLAGS}"
+		cxx_flags="-std=c++11"
+		debug="no"
+		FORTRANFLAGS="${CXXFLAGS}"
+		use_pch=$(usex pch)
+## In some cases other order can break the detection of right location of Boost: ##
+		system_fmt="y"
+		system_sundials="y"
+		system_eigen="y"
+		extra_inc_dirs="/usr/include/eigen3"
+	)
+	use test || scons_vars+=( googletest="none" )
+
+	scons_targets=(
+		f90_interface=$(usex fortran y n)
+		python2_package="none"
+	)
+
+	if use cti ; then
+		python_setup
```

Comment:
`python_setup` should already have been called by implicit `python-single-r1_pkg_setup`.

Reply:
Done.